packages/oauth2/src/common/jwt/__tests__/decode-jwt.test.ts

```diff
@@ -0,0 +1,155 @@
+import { describe, expect, test } from 'vitest'
+import { jwtSignerFromJwt } from '../decode-jwt'
+
+describe('Decode JWT', () => {
+  describe('jwtSignerFromJwt', () => {
+    test('x5c header', () => {
+      expect(jwtSignerFromJwt({ header: { x5c: ['cert'], alg: 'ES256' }, payload: {} })).toEqual({
+        method: 'x5c',
+        x5c: ['cert'],
+        alg: 'ES256',
+      })
+    })
+
+    test('did kid header', () => {
+      expect(jwtSignerFromJwt({ header: { kid: 'did:web:example.com#123', alg: 'ES256' }, payload: {} })).toEqual({
+        method: 'did',
+        didUrl: 'did:web:example.com#123',
+        alg: 'ES256',
+      })
+    })
+
+    test('did kid header with iss', () => {
+      expect(
+        jwtSignerFromJwt({ header: { kid: '#123', alg: 'ES256' }, payload: { iss: 'did:web:example.com' } })
+      ).toEqual({
+        method: 'did',
+        didUrl: 'did:web:example.com#123',
+        alg: 'ES256',
+      })
+    })
+
+    test('did in kid mismatch with iss', () => {
+      expect(() =>
+        jwtSignerFromJwt({
+          header: { kid: 'did:web:example.nl#123', alg: 'ES256' },
+          payload: { iss: 'did:web:example.com' },
+        })
+      ).toThrow(
+        `Unable to extract signer method from jwt. Found 1 allowed signer method(s) but contained invalid configuration:
+❌ method did: kid in header starst with did that is different from did value in 'iss'`
```

Comment:
Emojis in errors?

Reply:
Yeah i was doubting. Should i remove them?

Reply:
Removed them